### PR TITLE
Add capi v1alpha3 

### LINKED ISSUE #168: Add v1alpha3 API types
We already cut a release for v1alpha2 API types in v0.3.x which compatible with capi v0.2.x. We need to add v1alpha3 types as the first step to make capdo also compatible with capi v0.3.x.

- Copy types from api/v1alpha2.
- Rewrite imports.
- Set the storage version to v1alpha3 types.
- Make sure the generated CRDs contain both v1alpha2 and v1alpha3.

For the detail about migrating v1alpha2 to v1alpha3 can be found [here](https://cluster-api.sigs.k8s.io/developer/providers/v1alpha2-to-v1alpha3.html)

/kind feature
/milestone v0.4
/priority important-soon

### PR DESCRIPTION
**Pull request by prksu**
**What this PR does / why we need it**:
This PR adding support cluster api v1alpha3. 

**Which issue(s) this PR fixes**
Fixes #168

**Special notes for your reviewer**:
This implementation based on https://cluster-api.sigs.k8s.io/developer/providers/v1alpha2-to-v1alpha3.html guide

**Release note**:

```release-note
Support cluster api v1alpha3 types
```

### CODE REVIEW

**Comment by cpanato** [comment]
@prksu sorry for the delay, i will review this late today or tomorrow my morning.

File: config/manager/manager_image_patch.yaml [hunk 1]
```diff
@@ -8,6 +8,5 @@ spec:
     spec:
       containers:
       # Change the value of image field below to your controller image URL
-      - image: cluster-api-do-controller
+      - image: gcr.io/k8s-staging-cluster-api-do/cluster-api-do-controller
```
Comment (xmudrii) [inline config/manager/manager_image_patch.yaml:11]: Why we use staging instead of prod image here? For what's this image used exactly?

Comment (prksu) [reply config/manager/manager_image_patch.yaml:11]: The image here is a just default image so when some users/developers want to install this provider without building the image first they can use this manifest using a staging image (which is the current branch master state).

The other provider only uses production image in release manifest so i think we need to have a similar implementation.

Comment (xmudrii) [reply config/manager/manager_image_patch.yaml:11]: Sounds good!

**Pull request merged**